bionic/core/flow_execution.py

```diff
@@ -439,10 +478,15 @@ class TaskKeyLogger:
     def __init__(self, bootstrap):
         self._level = logging.INFO if bootstrap is not None else logging.DEBUG
 
-        executor = bootstrap.executor if bootstrap is not None else None
+        executor = bootstrap.process_executor if bootstrap is not None else None
         if executor is not None:
             self._already_logged_task_key_set = executor.create_synchronized_set()
         else:
+            # AIP would get the synchronized set, but the main process
```

Comment:
I'm not sure how to read this comment. The use of "would" suggests that we're talking about something hypothetical, but this seems like it's describing what will actually happen (in particular case).

It sounds like we're saying that some stuff won't work if AIP is enable. If so, let's be precise about
1. What circumstances will cause something weird to happen (AIP being enabled?)
2. Exactly what will happen (everything will work except that progress reporting will not include stuff happening on AIP?)
3. Whether this is something we need to come back and fix (TODO?)